### PR TITLE
Add observation endpoints to records

### PR DESCRIPTION
**Pull request by xjules**
This adds option to add observations to records and vice-versa.

### CODE REVIEW

File: ert_storage/endpoints/records.py [hunk 1]
```diff
@@ -324,6 +324,67 @@ async def get_record_metadata(
...
+    )
+    if observations:
+        record_obj.observations = observations
+        flag_modified(record_obj, "observations")
+        db.commit()
+    else:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail={
+                "error": f"Problem with linking observations {observation_ids} to record {name}!",
```
Comment (ManInFez) [inline ert_storage/endpoints/records.py:353]:
```suggestion
                "error": f"Observations {observatiion_ids} not found",
```

File: ert_storage/endpoints/records.py [hunk 2]
```diff
@@ -324,6 +324,67 @@ async def get_record_metadata(
...
+
+    observations = (
+        db.query(ds.Observation).filter(ds.Observation.id.in_(observation_ids)).all()
+    )
+    if observations:
+        record_obj.observations = observations
+        flag_modified(record_obj, "observations")
+        db.commit()
+    else:
+        raise HTTPException(
```
Comment (ManInFez) [inline ert_storage/endpoints/records.py:350]: Maybe you could also include a test that tries to link observations that are not there?

Comment (xjules) [reply ert_storage/endpoints/records.py:350]: Should be ready now!

**Review by ManInFez — APPROVED** [review]
LGTM

**Pull request merged**